sipXyealink/src/org/sipfoundry/sipxconfig/phone/yealink/YealinkPhone.java

```diff
@@ -215,6 +223,18 @@ public boolean getHasHD() {
     public String getDirectedCallPickupString() {
         return getPhoneContext().getPhoneDefaults().getDirectedCallPickupCode();
     }
+    
+    public String getPhonebookType(){
+        //Setting setting = getSettings().getSetting(PHONEBOOKTYPE);
+        Setting settings = getSettings();
+        Setting setting = settings.getSetting(PHONEBOOKTYPE);
+        String settingValue = (null == setting ? null : (setting.getValue() == null ? null : setting
+                        .getValue()));
+        if (!StringUtils.isEmpty(settingValue)) {
+                    return settingValue;
+                }
+        return "2";
```

Comment:
If you know how difficult it was for me to make it compile on Fedora23, right now i dont know if I could do it again...
Since I have to change the code I will work on that.